lxd/patches.go

```diff
@@ -924,26 +1056,31 @@ func upgradeFromStorageTypeZfs(name string, d *Daemon, defaultPoolName string, d
 			_, err := dbStoragePoolVolumeCreate(d.db, cs, storagePoolVolumeTypeContainer, poolID, volumeConfig)
 			if err != nil {
 				shared.LogWarnf("Could not insert a storage volume for snapshot \"%s\".", cs)
-				continue
```

Comment:
Same as for ZFS, we should check that the error is because of an existing DB entry, if not, fail.